core/primitives/src/optimistic_block.rs

```diff
@@ -0,0 +1,82 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_crypto::Signature;
+
+use crate::block::BlockHeader;
+use crate::hash::{hash, CryptoHash};
+use crate::merkle::combine_hash;
+use crate::stateless_validation::SignatureDifferentiator;
+use crate::types::BlockHeight;
+
+#[derive(BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
+pub struct OptimisticBlockInner {
+    signature_differentiator: SignatureDifferentiator,
+    pub block_height: BlockHeight,
+    pub block_timestamp: u64,
+    // Data to confirm the correctness of randomness beacon output
+    pub random_value: CryptoHash,
+    pub vrf_value: near_crypto::vrf::Value,
+    pub vrf_proof: near_crypto::vrf::Proof,
+}
+
+/// An optimistic block is independent of specific chunks and can be generated
+/// and distributed immediately after the previous block is processed.
+/// This block is shared with the validators and used to optimistically process
+/// chunks before they get included in the block.
+#[derive(BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
+#[borsh(init=init)]
+pub struct OptimisticBlock {
+    pub prev_block_hash: CryptoHash,
```

Comment:
Well, if we agree on using SignatureDifferentiator, prev_block_hash as a separate field doesn't give us anything, adding it to OptimisticBlockInner is simpler.